pkg/standalone-utils/test/VaultActionsStableJoinExit.test.ts

```diff
@@ -723,6 +724,237 @@ describe('Vault Actions - Stable Pools', () => {
     });
   });
 
+  describe('exit pool in recovery mode', () => {
+    const amountInBPT = fp(1);
+
+    async function getBPT(poolId: string): Promise<TokenList> {
+      return new TokenList([await Token.deployedAt(getPoolAddress(poolId))]);
+    }
+
+    context('when caller is not authorized', () => {
+      it('reverts', async () => {
+        await expect(
+          relayer.connect(other).multicall([
+            await encodeExitPool(vault, relayerLibrary, tokens, {
+              poolKind: PoolKind.COMPOSABLE_STABLE_V2,
+              poolId: poolIdStable,
+              userData: BasePoolEncoder.recoveryModeExit(amountInBPT),
+              toInternalBalance: true,
+              sender: user.address,
+              recipient,
+            }),
+          ])
+        ).to.be.revertedWith('Incorrect sender');
+      });
+    });
+
+    context('when caller is authorized', () => {
+      let sender: Account;
+
+      sharedBeforeEach('enter recovery mode', async () => {
+        await stablePool.enableRecoveryMode(admin);
+
+        expect(await stablePool.inRecoveryMode()).to.be.true;
+      });
+
+      context('sender = user', () => {
+        beforeEach(async () => {
+          sender = user;
+          await tokens.approve({ to: vault, from: user });
```

Comment:
Question: This would be just for the chained swaps right?

Reply:
Yes; if you comment it out, that specific test fails.